lib/tests/streamlit/runtime/state/session_state_test.py

```diff
@@ -317,6 +318,38 @@ def test_updates(self):
         assert sr5.get("slider")[0].value == 100.0
         assert sr5.get("slider")[1].value == 212.0
 
+    def test_serializable_check(self):
+        with patch_config_options({"runner.onlySerializableSessionState": True}):
+            script = self.script_from_string(
+                "unserializable.py",
+                """
+                import streamlit as st
+
+                def nested():
+                    return lambda x: x
+
+                st.session_state.unserializable = nested()
+                """,
+            )
+            sr = script.run()
+            assert sr.get("exception")
```

Comment:
TODO: when we have a custom error with descriptive text, let's assert that text here